### PR TITLE
WIP Add execution policy

### PR DESCRIPTION
**Pull request by Aertalon**
Use execution policies to parallelize gradient and hessian
computations.

### CODE REVIEW

**Comment by Aertalon** [comment]
std::transform is not constexpr if execution policies are used

File: src/convopt.hpp [hunk 1]
```diff
@@ -27,13 +28,17 @@ constexpr auto gradient(const P& p, F cost) -> distance_t<P>
...
-
-    for (std::size_t i{0U}; i < N; ++i) {
-        auto di = d;
-        di[i].imag = 1.0F;
-
-        r[i] = cost(di).imag;
-    }
+    auto is = std::views::iota(std::size_t{0}, N);
+    std::transform(
+        std::execution::par_unseq,
```
[the remark this answers is not in the document]
Comment (Aertalon) [reply src/convopt.hpp:33]: Does it parallelize in any way using the index sequences?

[the remark this answers is not in the document]
Comment (oliverlee) [reply src/convopt.hpp:33]: Not sure - haven't looked into it.

Nice trick with iota :)

**Review by oliverlee — APPROVED** [review]

**Comment by oliverlee** [comment]
looks like the clang builds need to be fixed?

**Comment by oliverlee** [comment]
This will need to be rebased off https://github.com/Aertalon/optimizer/pull/20

**Pull request merged**